dbms/src/Storages/S3/FileCache.h

```diff
@@ -227,8 +228,15 @@ class FileCache
 public:
 #endif
 
+    static FileCache & initializeInstance(
+        PathCapacityMetricsPtr capacity_metrics_,
+        const StorageRemoteCacheConfig & config_)
+    {
+        static FileCache instance(capacity_metrics_, config_);
+        return instance;
+    }
```

Comment:
why does it need to be changed from a static uniqut_ptr to a static instance?

Reply:
unique_ptr can't delete an incomplete type

Reply:
Reverted and add a new GlobalFileCacheInstance.